packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -36,9 +38,14 @@ class ErrorBadgeManager extends DisposableController
   }
 
   void _handleExtensionEvent(Event e) async {
-    // TODO(jacobr): badge inspector for appropriate errors.
     if (e.extensionKind == 'Flutter.Error') {
       incrementBadgeCount(LoggingScreen.id);
+
+      final json = e.extensionData.data;
+      final objectId = json['objectId'] as String;
+      if (objectId?.contains('inspector-') ?? false) {
```

Comment:
will this check include things we don't want to badge the inspector for? @jacob314

Reply:
All object ids sent with errors will start with `inspector-` currently so this won't really check for much. I think just assuming any case with an objectId is an inspector error is fine for now. We could add a TODO to verify that the objectId is referring to an Element or Widget (may require another round trip unless the json also contains the object type which it may.